src/widget/switch.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022 The Druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use glazier::kurbo::Circle;
+use vello::{
+    kurbo::{Point, Size},
+    peniko::Color,
+    SceneBuilder,
+};
+
+use crate::{IdPath, Message};
+
+use super::{
+    contexts::LifeCycleCx,
+    piet_scene_helpers::{fill_color, stroke},
+    AccessCx, BoxConstraints, ChangeFlags, Event, EventCx, LayoutCx, LifeCycle, PaintCx, UpdateCx,
+    Widget,
+};
+
+pub struct Switch {
+    id_path: IdPath,
+    is_on: bool,
+    is_moved: bool,
```

Comment:
(Slightly nit picky) I would name this something like `is_dragging` as I think it's more appropriate...